src/rdb.h

```diff
@@ -112,6 +112,7 @@ static_assert(RDB_VERSION < RDB_FOREIGN_VERSION_MIN || RDB_VERSION > RDB_FOREIGN
 #define RDB_TYPE_STREAM_LISTPACKS_2 19
 #define RDB_TYPE_SET_LISTPACK 20
 #define RDB_TYPE_STREAM_LISTPACKS_3 21
+#define RDB_TYPE_HASH_2 22
 /* NOTE: WHEN ADDING NEW RDB TYPE, UPDATE rdbIsObjectType(), and rdb_type_string[] */
```

Comment:
Did you do this? I can't spot the changes to rdbIsObjectType() and rdb_type_string[].

Reply:
maybe I missed them. Let me check if any farther changes are needed

Reply:
The `/* NOTE */` comment is placed here without any blank line to make it impossible to miss it. :smile: Still it doesn't catch people's attention. Fortunately I saw it when reviewing.

rdbIsObjectType is defined just 3 lines below this line.

I think we need do something else to make this impossible to miss.

```suggestion
#define RDB_END_OF_TYPES 23 /* Remember to update rdb_type_string[] when bumping this */
```

and

```diff
 /* Test if a type is an object type. */
-#define rdbIsObjectType(t) (((t) >= 0 && (t) <= 7) || ((t) >= 9 && (t) <= 21))
+#define rdbIsObjectType(t) (((t) >= 0 && (t) <= 7) || ((t) >= 9 && (t) < RDB_END_OF_TYPES))
```

and something for `rdb_type_string` to static assert that `rdb_type_string[RDB_END_OF_TYPES] == "(end of types)"` or something that would fail...

Reply:
done